src/binder/activitytracker.cpp

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ============================================================
+//
+// activitytracker.cpp
+//
+
+
+//
+// Helpers for interaction with the managed ActivityTracker
+//
+// ============================================================
+
+#include "common.h"
+#include "activitytracker.h"
+
+void ActivityTracker::Start(const WCHAR *providerName, const WCHAR *activityName, /*out*/ GUID *activityId, /*out*/ GUID *relatedActivityId)
+{
+    GCX_COOP();
+    struct _gc {
+        OBJECTREF activityTracker;
+        STRINGREF providerName;
+        STRINGREF activityName;
+    } gc;
+    ZeroMemory(&gc, sizeof(gc));
+
+    GCPROTECT_BEGIN(gc);
+
+    MethodDescCallSite getInstance(METHOD__ACTIVITY_TRACKER__GET_INSTANCE);
+    gc.activityTracker = getInstance.Call_RetOBJECTREF(NULL);
+    gc.providerName = StringObject::NewString(providerName);
```

Comment:
It looks like the providerName and activityName is always the same. It may be nice to not pass it to managed side at all; or if it is going to be a set of fixed values - pass it to the managed side as enum so that the string does not need to be allocated each time.

Reply:
Made the managed side always use the same string, since this is the only one using it and with the same strings right now.